node/src/rpc/rpc_reducer.rs

```diff
@@ -79,11 +80,21 @@ impl RpcState {
                 state.requests.insert(*rpc_id, rpc_state);
 
                 let dispatcher = state_context.into_dispatcher();
+
+                let callback = redux::callback!(
+                    on_p2p_connection_outgoing_rpc_connection_success((peer_id: PeerId, rpc_id: Option<RpcId>)) -> crate::Action {
+                        let Some(rpc_id) = rpc_id else {
+                            panic!("RPC ID not provided");
```

Comment:
is there any way to avoid this panic here?

Reply:
I could add enum callback like for `P2pNetworkKademliaStreamWaitOutgoingCallback`

Reply:
@0xMimir maybe it is not that bad being that the `rpc_id` argument is immediately after this code (or immediately before if you inline the callback). So it could be made into an `unreachable!` with a comment explaining why. I think I did something similar with other callbacks